File: src/FizzBuzz Variety Sample Codes/fizzbuzz2.rb
# FizzBuzzクラスを定義 --- (*1)
class FizzBuzz
  # 初期化メソッド --- (*2)
  def initialize(max)
    @max = max
    # FizzとBuzzの判定条件を定義 --- (*3)
    @is_fizz = lambda { |n| n % 3 == 0 }
    @is_buzz = lambda { |n| n % 5 == 0 }
  end
  # 繰り返しFizzBuzzの値を出力 --- (*4)
  def run
    1.upto(@max){ |n| puts check(n) }
  end
  # FizzBuzzの値を判定 --- (*5)
  def check(n)
    if @is_fizz.call(n) and @is_buzz.call(n)
      "FizzBuzz"
    elsif @is_fizz.call(n)
      "Fizz"
    elsif @is_buzz.call(n)
      "Buzz"
    else
      n.to_s
    end
  end
end

# クラスをインスタンス化して実行 --- (*6)
fb = FizzBuzz.new(100)
fb.run

File: src/FizzBuzz Variety Sample Codes/fizzbuzz2.cs
class FizzBuzz { // --- (*1)
  int max;
  FizzBuzz(int max) { // --- (*2)
    this.max = max;
  }
  void Run() { // --- (*3)
    for (int i = 1; i <= max; i++) {
      System.Console.WriteLine(this.Check(i));
    }
  }
  string Check(int i) { // --- (*4)
    if (i % 3 == 0 & i % 5 == 0) return "FizzBuzz";
    if (i % 3 == 0) return "Fizz";
    if (i % 5 == 0) return "Buzz";
    return i.ToString();

  }
  public static void Main() { // --- (*5)
    FizzBuzz obj = new FizzBuzz(100);
    obj.Run();
  }
}

